### ISSUE TITLE
Version ansible-lint 6.8.6 using ansible 2.13.5 broke

### ISSUE THREAD

**Issue by maxamillion** [ansible/ansible-lint#2673]
##### Summary
```
$(ansible)  ansible-lint --version
ansible-lint 6.8.6 using ansible 2.13.5

$(ansible)  ansible-lint
WARNING  Retrying execution failure 250 of: ansible-galaxy collection install -v --force -p /home/admiller/.cache/ansible-compat/42cc84/collections .
ERROR    Command returned 250 code:
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'osbuild.composer:0.0.1' to '/home/admiller/.cache/ansible-compat/42cc84/collections/ansible_collections/osbuild/composer'
to see the full traceback, use -vvv

ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: 'MANIFEST.json'

WARNING  Retrying execution failure 250 of: ansible-galaxy collection install -v --force -p /home/admiller/.cache/ansible-compat/42cc84/collections .
ERROR    Command returned 250 code:
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'osbuild.composer:0.0.1' to '/home/admiller/.cache/ansible-compat/42cc84/collections/ansible_collections/osbuild/composer'
to see the full traceback, use -vvv

ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: 'osbuild_setup_server.yml'

WARNING  Retrying execution failure 250 of: ansible-galaxy collection install -v --force -p /home/admiller/.cache/ansible-compat/42cc84/collections .
ERROR    Command returned 250 code:
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'osbuild.composer:0.0.1' to '/home/admiller/.cache/ansible-compat/42cc84/collections/ansible_collections/osbuild/composer'
to see the full traceback, use -vvv

ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: 'vars'

Command returned 250 code:
No config file found; using defaults
Starting galaxy collection install process
Process install dependency map
Starting collection install process
Installing 'osbuild.composer:0.0.1' to '/home/admiller/.cache/ansible-compat/42cc84/collections/ansible_collections/osbuild/composer'
to see the full traceback, use -vvv

ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: 'MANIFEST.json'
```

However, ansible-lint 6.8.4 works just fine:

```
$(ansible)  ansible-lint --version
ansible-lint 6.8.4 using ansible 2.13.5
A new release of ansible-lint is available: 6.8.4 → 6.8.6 Upgrade by running: pip install --upgrade ansible-lint

$(ansible)  ansible-lint
WARNING  Listing 61 violation(s) that are fatal
yaml[indentation]: Wrong indentation: expected 2 but found 0
galaxy.yml:20

yaml[comments]: Missing starting space in comment
galaxy.yml:30

yaml[comments]: Missing starting space in comment
galaxy.yml:31

.... OUTPUT OMITTED FOR BREVITY .....
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version

ansible-lint --version

```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### OS / ENVIRONMENT

Fedora 36
Python 3.10.7

##### STEPS TO REPRODUCE

When running ansible-lint version 6.8.6, do the following:

```console (paste below)
mkdir -p ~/ansible_collections/osbuild/
cd ~/ansible_collections/osbuild/
git clone https://github.com/ansible-collections/osbuild.composer composer
cd composer
ansible-lint

```

##### Desired Behavior

ansible-lint to work :)

##### Actual Behavior

listed above

**Comment by gygitlab** [comment]
I'm also seeing this. It's like the process is trying to install the galaxy collection of the repository itself.

Offline mode seems to workaround for now.

**Comment by ssbarnea** [comment]
@maxamillion Based on some tests, this error might come from Python itself but I proposed https://github.com/ansible/ansible/pull/79409 to make the rmdir bit more resilient to such errors.

**Comment by ssbarnea** [comment]
Please let me know if you can reproduce the problem using the newer 6.9.0 release.

**Comment by SirUli** [comment]
@ssbarnea since i faced the same in #2651 i can at least report that this is fixed for me with v6.9.0. Thanks for your help on this!
